src/api/firebase.js

```diff
@@ -92,16 +93,77 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 export async function updateItem(listId, itemId, checked) {
 	const itemRef = doc(db, listId, itemId);
 	const itemDoc = await getDoc(itemRef);
-	const { dateLastPurchased, totalPurchases } = itemDoc.data();
+	const { dateCreated, dateLastPurchased, dateNextPurchased, totalPurchases } =
+		itemDoc.data();
 
-	try {
-		return updateDoc(itemRef, {
-			dateLastPurchased: checked ? new Date() : dateLastPurchased,
-			totalPurchases: checked ? totalPurchases + 1 : totalPurchases,
-			checked: checked,
-		});
-	} catch (error) {
-		console.log('updateDoc error', error);
+	const checkedDateLastPurchased = checked ? new Date() : dateLastPurchased;
+
+	if (checkedDateLastPurchased) {
+		// Convert dateNextPurchased from Timestamp to a JS Date
```

Comment:
Comments can be useful, but be mindful of "over-commenting" and potentially cluttering your code. This variable has been named perfectly, and I understood what you were without even reading the comments

Reply:
@mentalcaries The comment was deleted as requested. Your feedback is appreciated!